File: src/main/java/com/vubq/fashionstorewebsite/entities/Color.java
package com.vubq.fashionstorewebsite.entities;

import com.vubq.fashionstorewebsite.dtos.ColorDto;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.FieldNameConstants;
import lombok.experimental.SuperBuilder;
import org.hibernate.annotations.UuidGenerator;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@FieldNameConstants
@SuperBuilder
@Entity
@Table(name = "colors")
public class Color extends BaseEntity {

    @Id
    @UuidGenerator
    @Column(name = "id")
    private String id;

    @Column(name = "name")
    private String name;

    @Column(name = "description")
    private String description;

    public static Color toEntity(ColorDto colorDto) {
        return Color.builder()
                .id(colorDto.getId())
                .name(colorDto.getName())
                .description(colorDto.getDescription())
                .createdBy(colorDto.getCreatedBy() != null ? colorDto.getCreatedBy().getId() : null)
                .updatedBy(colorDto.getUpdatedBy() != null ? colorDto.getUpdatedBy().getId() : null)
                .createdAt(colorDto.getCreatedAt())
                .status(colorDto.getStatus())
                .updatedAt(colorDto.getUpdatedAt())
                .build();
    }
}

File: src/main/website/models/Color.ts
import {Status} from '~/models/enums/Status'
import {CreatedBy} from '~/models/CreatedBy'
import {UpdatedBy} from '~/models/UpdatedBy'

export class Color {
  public id?: string
  public name?: string
  public description?: string
  public createdBy?: CreatedBy
  public updatedBy?: UpdatedBy
  public createdAt?: Date
  public updatedAt?: Date
  public status?: Status

  constructor(init?: Partial<Color>) {
    Object.assign(this, init)
  }
}
